### PR TITLE
feat(api): add historical market prices and expose in api

### PR DESCRIPTION
**Pull request by daywiss**
Signed-off-by: David Adams <david@umaproject.org>

**Motivation**
https://app.clubhouse.io/uma-project/story/1673/add-historical-prices-for-market-data

**Summary**

Adds a table to accumualte historical prices for market prices from the time server starts. This will not backfill history, its not clear that its possible with the api we are using. 

**Details**

This exposes 2 new api endpoints:
- historicalMarketPricesBetween
- sliceHistoricalMarketPrices

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [x]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

**Issue(s)**

https://app.clubhouse.io/uma-project/story/1673/add-historical-prices-for-market-data

### CODE REVIEW

File: packages/api/src/services/actions.ts [hunk 1]
```diff
@@ -140,6 +140,15 @@ export function Handlers(config: Config, appState: Dependencies): Actions {
     async listTvms(currency: CurrencySymbol = "usd") {
       return appState.stats[currency].latest.tvm.values();
     },
+    async historicalMarketPricesBetween(tokenAddress: string, start = 0, end: number = nowS()) {
+      assert(tokenAddress, "requires token address");
+      return marketPrices.usdc.history.betweenByAddress(tokenAddress, start, end);
+    },
+    async sliceHistoricalMarketPrices(tokenAddress: string, start = 0, length = 1) {
+      assert(tokenAddress, "requires token address");
+      assert(length < 1000, "length must be less than 1000 samples");
```
Comment (nicholaspai) [inline packages/api/src/services/actions.ts:149]: Why this magic number?

**Review by nicholaspai — APPROVED** [review]
LGTM

**Review by mrice32 — APPROVED** [review]
LGTM!

**Review by chrismaree — APPROVED** [review]

File: packages/api/src/services/actions.ts (hunk 1 above)
Comment (daywiss) [reply packages/api/src/services/actions.ts:149]: good question, i think chris brought up a while ago we should probably limit how many samples get returned from api.  This caps it at 1000. should probably change this into a config var though!

Comment (daywiss) [reply packages/api/src/services/actions.ts:149]: going to save this change for a chore ticket, its the last task here https://app.clubhouse.io/uma-project/story/1636/various-api-improvements

**Pull request merged**